### DISCUSSION TITLE
Make type internal to Namespace

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#5884] [Language Ideas] [answered]
It would be great if there was a way to make a type internal so it was only visible to types in the same assembly in the name namespace. 

I had the idea:

//This makes anything in this block not visible to other namespaces. 
internal namespace System.Collections.Generic {
  public class Foo{}
}

**Comment by PathogenDavid** [comment, marked as answer]
See https://github.com/dotnet/csharplang/issues/2497 and https://github.com/dotnet/csharplang/discussions/1138

You could enforce this pretty easily with an analyzer.
